ecologits/model_repository.py

```diff
@@ -31,16 +31,19 @@ class Model:
 
 
 class ModelRepository:
-
     def __init__(self, models: list[Model]) -> None:
         self.__models = models
 
     def find_model(self, provider: str, model_name: str) -> Optional[Model]:
-        for model in self.__models:
-            # To handle specific LiteLLM calling (e.g., mistral/mistral-small)
-            if model.provider == provider and model.name in model_name:
-                return model
-        return None
+        provider_models = [model for model in self.__models if model.provider == provider]
+        return next(
+            (
+                model
+                for model in provider_models
+                if (model.name == model_name or model.name.replace(".", "") == model_name)
+            ),
+            next((model for model in provider_models if model.name in model_name), None),
+        )
```

Comment:
Instead of changing this function, I would rather add the models to the models.csv file. For instance, `gpt-35-turbo` will be a duplicated line of `gpt-3.5-turbo`. We plan to refactor the model repository to introduce aliases in the future.

Reply:
Ok perfect, adding some kind of aliases mechanic was something I tought of but felt was a bit overkill for just azure.

This function still needs to be changed at some point though. Currently it seems like gpt-4-turbo and gpt-o match with gpt-4. (I can move this to another PR/open an issue if needed as this is a separated problem).

Reply:
Agree, I think this bug was introduced when adding litellm, will get it fixed in the next release for sure! 🙏

Reply:
Adding the tests is proving to be a bit more complicated than I anticipated.

Python-recording or the underlying vcrpy seems to always be trying to overwrite some cassettes.
Running "make test" raises `vcr.errors.CannotOverwriteExistingCassetteExceptionon` on google, huggingface and litellm tests. Even after cloning and reinstalling from scratch this still happens.

I did manage to generate a cassette for azure openai, but whenever I try to run the tests using it, the same error is raised.

Any idea on how to solve this?

Reply:
Ok thanks for letting me know. That is indeed a strange behavior... Can you push your code as it is right now? So I can try on my side, see if I have the same issue.

Reply:
@NP4567-dev I think I have found your work on your fork (branch `feature/azure-tests`). 

The issue you have is probably because the name of the tests you've added for Azure OpenAI are equal to the tests for OpenAI.

Meaning that if you change the function names from `test_openai_chat` to `test_azure_openai_chat` in the `test_azure_openai.py` file, it should work properly. :)

Reply:
Yeah, this was another problem I had already fixed locally.
I did find the issue though, we have a proxy that prevented url matching on the cassettes, hence the recording attempt.
It is fixed now, I just have a few final verification to make and it should be ready to merge